lib/seattleflu/db/cli/command/longitudinal.py

```diff
@@ -0,0 +1,403 @@
+    except ValueError:
+        raise Exception(f"""Expected the «{time}» column to contain only
+            integers after casting longitudinal data from wide-to-long.
+            Please check your stubnames and try again.""")
+
+    return reshaped
+
+
+def rename_stub(df: pd.DataFrame, regex: str, current_stub: str,
+                desired_stub: str) -> pd.DataFrame:
+    """
+    Renames *current_stub* columns in a given DataFrame *df* to a
+    *desired_stub*, using the given *regex* to capture the desired columns
+    to rename.
+
+    Motivation: useful for avoiding future DataFrame reshaping conflicts with
+    the original stubname.
+
+    For example, the stub "swab" in the original data is not sufficient for
+    capturing longitudinal data only related to "swab_x" columns; other distinct
+    "swab_*" stubs like "swab_date_x" would be artifically captured by the
+    "stub" swabname if the original stub were not converted to a longer or more
+    unique stubname like "swab_collected_x".
+    """
+    rename_map = {}
+
+    stub_columns = list(df.filter(regex=regex, axis=1))
+
+    for col in stub_columns:
+        rename_map[col] = re.sub(current_stub, desired_stub, col)
+
+    return df.rename(columns=rename_map)
+
+
+def longitudinal_stubnames(df: pd.DataFrame) -> [str]:
+    """
+    Returns a list of the stubnames of longitudinal columns that have suffixes
+    indicating the week number of the study (e.g. 'swab_date_11').
+    """
+    stubnames = []
+    for column in list(df):
+        match = re.match(".*(?=_[0-9]+$)", column)
+        if match:
+            stubnames.append(match[0])
+
+    return list(set(stubnames))
+
+
+def select_all_that_apply_wide_to_long(df: pd.DataFrame) -> pd.DataFrame:
+    """
+    Given a wide DataFrame *df*, collapses "Select All That Apply"-type
+    categorical survey responses with pseudo-one-hot-encoding format and
+    converts the responses into human-readable lists of equivalent logical
+    value.
+
+    Throws a :class:`AssertionError` if the number of rows in the resulting
+    DataFrame differs from the given *df*.
+
+    Contains hard-coded column names that may need to be updated in the future.
+    """
+
+    stubnames = [
+        "race", "feeding", "meds", "rn_sx", "sx_specific", "missed_impact",
+        "household_sx", "travel_type",
+    ]
+    pid = ['study_id'] + ['week']
+
+    stubbed_columns = list(df.filter(regex='|'.join(stubnames)))
+    reshaped_data = df.drop(stubbed_columns, axis=1)
+
+    for stub in stubnames:
+        true_values = collapse_wide_stubbed_columns(df, stub, pid)
+        if true_values is None:
+            continue
+
+        reshaped_data = reshaped_data.merge(true_values, how='left', on=pid)
+
+    assert len(df) == len(reshaped_data), f"You do not have a 1:1 merge on {pid}"
+
+    return reshaped_data
+
+
+def collapse_wide_stubbed_columns(df: pd.DataFrame, stub: str,
+                                  pid: [str]) -> pd.DataFrame:
```

Comment:
FYI, I think you want `List[str]` here, but again, I know this isn't clear because we're not validating these type annotations yet.